yeonjae/baekjoon/스택/4949-균형잡힌세상.js

```diff
@@ -0,0 +1,39 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const strList = input.slice(0, input.length - 1);
+  let result = "";
+  strList.forEach((str) => {
+    const stack = [];
+    const bracketObj = {
+      ")": "(",
+      "]": "[",
+    };
+    const bracketValues = Object.values(bracketObj);
+    const bracketKeys = Object.keys(bracketObj);
+
+    for (const char of str) {
+      if (bracketValues.includes(char)) {
+        stack.push(char);
+      }
+      if (bracketKeys.includes(char)) {
+        if (stack.at(-1) === bracketObj[char]) {
+          stack.pop();
+        } else {
+          stack.push(char);
+        }
+      }
+    }
+    result += stack.length > 0 ? "no\n" : "yes\n";
+  });
+  console.log(result);
```

Comment:
```suggestion
  const strList = input.slice(0, input.length - 1);
  const bracketObj = {
    ")": "(",
    "]": "[",
  };
  const bracketValues = Object.values(bracketObj);
  const bracketKeys = Object.keys(bracketObj);
  let result = "";
  
  strList.forEach((str) => {
    const stack = [];

    for (const char of str) {
      if (bracketValues.includes(char)) {
        stack.push(char);
      }
      if (bracketKeys.includes(char)) {
        if (stack.at(-1) === bracketObj[char]) {
          stack.pop();
        } else {
          stack.push(char);
        }
      }
    }
    result += stack.length > 0 ? "no\n" : "yes\n";
  });
  console.log(result);
```
- 풀이잘봤습니다~ 🙇‍♂️
- 저의 풀이와 유사한데 스스로 개선을 해보자면 ㅋㅋ bracketObj, bracketValues, bracketKeys는 for...of문 외부에다 선언해주는게 좋을 것 같습니다 :)